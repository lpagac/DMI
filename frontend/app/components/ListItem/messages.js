/*
 * ListItem Messages
 *
 * This contains all the text for the ListItem component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ListItem';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ListItem component!',
  },
});
