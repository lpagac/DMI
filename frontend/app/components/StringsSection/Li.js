import styled from 'styled-components';

const Li = styled.li`
  display: inline;
  margin: 0 15px;
`;

export default Li;
