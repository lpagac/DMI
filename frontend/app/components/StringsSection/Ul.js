import styled from 'styled-components';

const Ul = styled.ul`
  list-style: none;
  text-align: center;
  margin: 3em auto;
  line-height: 200%;
`;

export default Ul;
