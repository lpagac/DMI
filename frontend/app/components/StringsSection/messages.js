/*
 * StringsSection Messages
 *
 * This contains all the text for the StringsSection component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.StringsSection';

export default defineMessages({
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Uh-Oh, No strings found. Try again later.',
  },
});
