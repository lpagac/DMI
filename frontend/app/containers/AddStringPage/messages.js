/*
 * AddStringPage Messages
 *
 * This contains all the text for the AddStringPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.AddStringPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Add A String!',
  },
  invalidInput: {
    id: `${scope}.invalidInput`,
    defaultMessage: 'Try adding some text!',
  },
  submit: {
    id: `${scope}.submit`,
    defaultMessage: 'ADD',
  },
  error: {
    id: `${scope}.error`,
    defaultMessage:
      'Unable to add string. Please check your input and try again.',
  },
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Nice string! Add some more!',
  },
});
