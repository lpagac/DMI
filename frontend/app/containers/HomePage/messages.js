/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Check out our latest strings!',
  },
  subHeader: {
    id: `${scope}.subHeader`,
    defaultMessage: 'Limited time only... Add your own string!',
  },
});
