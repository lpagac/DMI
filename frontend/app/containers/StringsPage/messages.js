/*
 * StringsPage Messages
 *
 * This contains all the text for the StringsPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.StringsPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Pretty sweet collection of strings right?',
  },
});
