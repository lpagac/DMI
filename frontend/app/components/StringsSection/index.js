import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import LoadingIndicator from 'components/LoadingIndicator';
import H3 from 'components/H3';
import Ul from './Ul';
import Li from './Li';
import messages from './messages';

function StringsSection({ loading, error, strings }) {
  if (loading) {
    return (
      <div data-testid="loading-indicator">
        <LoadingIndicator data-testid="loading-indicator" />
      </div>
    );
  }

  if (error !== false) {
    const ErrorComponent = () => (
      <H3>
        <FormattedMessage {...messages.error} />
      </H3>
    );
    return <ErrorComponent />;
  }

  if (strings !== false) {
    try {
      return (
        <Ul>
          {strings.map(({ id, string }) => (
            <Li key={id}>{string}</Li>
          ))}
        </Ul>
      );
    } catch (err) {
      return null;
    }
  }

  return null;
}

StringsSection.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.any,
  strings: PropTypes.any,
};

export default StringsSection;
