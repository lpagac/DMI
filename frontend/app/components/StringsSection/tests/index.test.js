/**
 *
 * Tests for StringsSection
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import StringsSection from '../index';
import messages from '../messages';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<StringsSection />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <StringsSection />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <StringsSection />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should display current strings', () => {
    const utils = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <StringsSection
          loading={false}
          error={false}
          strings={[{ id: 'test-id', string: 'test string' }]}
        />
      </IntlProvider>,
    );

    const stringsSection = utils.getByText('test string');
    expect(stringsSection).not.toEqual(null);
  });

  it('should display error message if error is true', () => {
    const utils = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <StringsSection loading={false} error strings={false} />
      </IntlProvider>,
    );

    const errorSection = utils.getByText(messages.error.defaultMessage);
    expect(errorSection).not.toEqual(null);
  });

  it('should display loading indicator if loading is true', () => {
    const utils = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <StringsSection loading error={false} strings={false} />
      </IntlProvider>,
    );

    const loadingSection = utils.queryByTestId('loading-indicator');
    expect(loadingSection).not.toEqual(null);
  });

  it('should display nothing if unable to map over strings', () => {
    const utils = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <StringsSection loading={false} error={false} strings="not an array" />
      </IntlProvider>,
    );

    const stringsSection = utils.queryByText('not an array');
    expect(stringsSection).toEqual(null);
  });
});
