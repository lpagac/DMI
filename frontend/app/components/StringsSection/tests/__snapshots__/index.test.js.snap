// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringsSection /> Should render and match the snapshot 1`] = `
<h3
  class="H3-sc-1nwhpi5-0 kNbuJo"
>
  <span>
    Uh-Oh, No strings found. Try again later.
  </span>
</h3>
`;
